lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -676,7 +676,10 @@ def stub_forge(forge_host = nil)
         def sleep_until_puppetdb_started(host, nonssl_port = nil, ssl_port = nil)
           nonssl_port = options[:puppetdb_port_nonssl] if nonssl_port.nil?
           ssl_port = options[:puppetdb_port_ssl] if ssl_port.nil?
-          curl_with_retries("start puppetdb", host, "http://localhost:#{nonssl_port}", 0, 120)
+          endpoint = 'status/v1/services/puppetdb-status'
+          retry_on(host,
+                   "curl -m 1 http://localhost:#{nonssl_port}/#{endpoint} | grep '\"state\":\"running\"'",
+                   {:max_retries => 120})
           curl_with_retries("start puppetdb (ssl)",
```

Comment:
@johnduarte I know this isn't part of your code change, but it seems to me that this `curl_with_retries` is not necessary. If the `retry_on` host command is successful, isn't puppetdb ostensibly listening to the ssl port as well? What do you think about removing that `curl_with_retries` after the `retry_on`?